File: java-poc/src/main/java/io/craigmiller160/querydslpoc/java/data/entity/Employee.java
package io.craigmiller160.querydslpoc.java.data.entity;

import java.time.LocalDate;
import java.util.UUID;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "employees")
public class Employee {
  @Id private UUID id;
  private String firstName;
  private String lastName;
  private LocalDate dateOfBirth;
  private UUID positionId;
}

File: kotlin-poc/src/main/kotlin/io/craigmiller160/querydslpoc/kotlin/data/entity/Employee.kt
package io.craigmiller160.querydslpoc.kotlin.data.entity

import java.time.LocalDate
import java.util.UUID
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "employees")
class Employee {
    @Id
    lateinit var id: UUID
    lateinit var firstName: String
    lateinit var lastName: String
    lateinit var dateOfBirth: LocalDate
    lateinit var positionId: UUID
}